src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala

```diff
@@ -486,6 +489,133 @@ trait BronzeTransforms extends SparkSessionWrapper {
     }
   }
 
+  def prepClusterSnapshot(
+                           workspace: Workspace,
+                           pipelineSnapTS: TimeTypes,
+                           apiEnv: ApiEnv,
+                           database: Database,
+                           clusterSnapshotErrorsTarget: PipelineTable,
+                           config: Config,
+                           apiEndpointTempDir: String
+                         )(auditDF: DataFrame) : DataFrame = {
+
+    val clusterIDs = auditDF
```

Comment:
I think you could make this more efficient if you write a new function that extracts out the clusterIDs for both jobs and cluster services, instead of relying on existing instrumentation to get those. As it is now you would do 2 passes on the audit log data set, which we can reduce to 1

Reply:
@gueniai  I have tried this at very start only. Bur problem is for JobClusterIDs we are using WIndow function which we are not using for Clusters service name. Thats why have to use bot of this functions.

Reply:
but for this table you don't need a window function for this table, I can understand for the jobs table why we need it, but for this one a select distinct is enough